azure/functions/decorators/function_app.py

```diff
@@ -1326,140 +1463,104 @@ def decorator():
 
         return wrap
 
-    def generic_trigger(self,
-                        arg_name: str,
-                        type: str,
-                        data_type: Optional[Union[DataType, str]] = None,
-                        **kwargs
-                        ) -> Callable:
+
+class FunctionRegister(DecoratorApi, HttpFunctionsAuthLevelMixin, ABC):
+    def __init__(self, auth_level: Union[AuthLevel, str], *args, **kwargs):
+        """Interface for declaring top level function app class which will
+        be directly indexed by Python Function runtime.
+
+        :param auth_level: Determines what keys, if any, need to be present
+        on the request in order to invoke the function.
+        :param args: Variable length argument list.
+        :param kwargs: Arbitrary keyword arguments.
         """
-        The generic_trigger decorator adds :class:`GenericTrigger`
-        to the :class:`FunctionBuilder` object for building :class:`Function`
-        object used in worker function indexing model.
-        This is equivalent to defining a generic trigger in the
-        function.json which triggers function to execute when generic trigger
-        events are received by host.
-        All optional fields will be given default value by function host when
-        they are parsed by function host.
+        DecoratorApi.__init__(self, *args, **kwargs)
+        HttpFunctionsAuthLevelMixin.__init__(self, auth_level, *args, **kwargs)
 
-        Ref: https://aka.ms/azure-function-binding-custom
+    def get_functions(self) -> List[Function]:
+        """Get the function objects in the function app.
 
-        :param arg_name: The name of trigger parameter in the function code.
-        :param type: The type of binding.
-        :param data_type: Defines how Functions runtime should treat the
-         parameter value.
-        :param kwargs: Keyword arguments for specifying additional binding
-        fields to include in the binding json.
+        :return: List of functions in the function app.
+        """
+        return [function_builder.build(self.auth_level) for function_builder
+                in self._function_builders]
 
-        :return: Decorator function.
+    def register_functions(self, function_container: DecoratorApi) -> None:
```

Comment:
@YunchuWang - As this is only used for registering blueprints, should be call is `register_blueprint`? or you are using this to register for other other generic cases? We can use pointers to functions and use specific handlers for such cases.

Define a generic _register_functions
```
def _register_functions(...):
 ...
```

and then 
```
register_blueprint=_register_functions
register_wsgiapp=_register_functions
...
```

such that it is easier to understand what the method would actually be used for.

Reply:
it can be used to register custom defined classes extending triggerapi or bindingapi or both. i am thinking aliasing a register_blueprint and keeping register_functions public to register any other types of custom defined function container classes?